File: src/java/WebNinja/src/main/java/webninja/UserRepository.java
package webninja;

import java.util.HashMap;
import java.util.Map;

public class UserRepository {

    private final Map<String, User> users;

    public UserRepository() {
        users = new HashMap<String, User>();
        add(new User("Ninja1","n 1"));
        add(new User("Ninja2","n2 n2"));
    }

    private void add(User user) {
        users.put(user.getUseId(), user);
    }

    public User getUserById(String userId) {
        return users.get(userId);
    }
}

File: src/dotnet/WebNinja/WebNinja/webninja/UserRepository.cs


using System.Collections.Generic;

namespace WebNinja.webninja
{    
    public class UserRepository
    {
        private Dictionary<string, User> users;

        public UserRepository ()
        {
            users = new Dictionary<string, User>();
            Add(new User("Ninja 1","Ninja1"));
            Add(new User("Ninja 2", "Ninja2"));
        }

        public void Add(User user)
        {
            users.Add(user.UserId, user);
        }

        public User FindByUserId(string userId)
        {
            return users[userId];
        }
    }
}
